api/chain.yml

```diff
@@ -0,0 +1,307 @@
+  executed_message_resource:
+    get:
+      operationId: getExecutedMessageById
+      summary: Find ExecutedMessage by ID
+      description: Return information about the ExecutedMessage with the specified ID. An executed Message is one that has been included in a valid block in the chain. It includes return value and status information.
+      tags:
+        - Messages
+      parameters: 
+      - name: executedMessageId
+        in: path
+        description: ID of the executed message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/ExecutedMessage'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  message_resource:
+    get:
+      operationId: getMessageById
+      summary: Find Message by ID
+      description: Return information about the Message with the specified ID
+      tags: 
+        - nessages
+      parameters: 
+      - name: messageId
+        in: path
+        description: ID of the message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Message'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  messages_collection:
+    post:
+      operationId: createMessage
+      summary: Send a Message
+      description: This method sends a new Message to an Actor recipient.
+      tags: 
+        - Messages
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/models/Message'
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Message'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  tipset_resource:
+    get:
+      operationId: getTipsetById
+      summary: Fetch a Tipset by ID
+      description: Return information about the Tipset with the specified ID
+      tags:
+        - Tipsets
+      parameters: 
+      - name: tipsetId
+        in: path
+        description: ID of the Tipset to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Tipset'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  tipsets_collection:
+    get:
+      operationId: listTipsets
+      summary: List Tipsets
+      description: List Tipsets currently known to this node in reverse height order
+      tags:
+        - Tipsets
+      parameters:
```

Comment:
Add optional parameters of the tipset from which to start (default the implicit head), and the height at which to stop (default 0).